cmd/synthetic-monitoring-agent/secret.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+// Secret represents a string that shouldn't be logged.
+type Secret string
+
+// String returns a redacted version of the secret.
+func (s Secret) String() string {
+	return "<redacted>"
+}
+
+// Set sets the value of the secret.
+func (s *Secret) Set(value string) error {
+	*s = Secret(value)
+	return nil
+}
```

Comment:
Does `Set(string) error` make this implement any interface?

Reply:
ah, yeah, it's what flag.Value wants. It _also_ wants String() for that matter.

Reply:
It's needed for line 83 to work.